cli/src/commands/log.rs

```diff
@@ -38,8 +38,9 @@ use crate::ui::Ui;
 /// rendered as a synthetic node labeled "(elided revisions)".
 #[derive(clap::Args, Clone, Debug)]
 pub(crate) struct LogArgs {
-    /// Which revisions to show. Defaults to the `revsets.log` setting, or
-    /// `@ | ancestors(immutable_heads().., 2) | trunk()` if it is not set.
+    /// Which revisions to show. If no paths nor revisions are specified, this
+    /// defaults to the `revsets.log` setting, or `@ |
+    /// ancestors(immutable_heads().., 2) | trunk()` if it is not set.
```

Comment:
I would reword this slightly:

> Which revisions to show. If neither revisions nor paths are specified, the set of revisions defaults is the `revset.log` setting (if set) or `@ |ancestors(immutable_heads().., 2) | trunk()`. If any paths are specified, the default is `all()`. 

I think it could be improved further -- I'd like the short help to be, well, shorter, but I haven't come up with a great way to word that. If we only leave the first sentence in the short help, it will be easy to think that the argument is required or misunderstand the default.

Reply:
Perhaps, the detailed doc can be moved up to the command level? Something like
* `jj log` by default ...
* if `-r REVISION` or `PATH` is specified, all visible revisions will be searched...

Reply:
So we leave this part for now?

Reply:
Yeah, I would address it separately.